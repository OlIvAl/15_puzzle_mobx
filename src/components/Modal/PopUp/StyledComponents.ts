import styled from 'styled-components';

export const StyledPopUpWrapper = styled.div`
  width: 300px;
  padding: 20px;
  background-color: #fff;
  display: flex;
  flex-direction: column;
  align-items: center;
`;
export const StyledPopUpHeader = styled.div`
  font-size: 24px;
  font-weight: bold;
  margin-bottom: 12px;
`;
export const StyledPopUpContent = styled.div`
  
`;
export const StyledPopUpCloseButton = styled.button`
    border-radius: 4px;
    padding: 6px 10px;
    background-color: #358ada;
    color: #fff;
    text-decoration: none;
    border: none;
    width: 50%;
    margin-top: 20px;
  `;