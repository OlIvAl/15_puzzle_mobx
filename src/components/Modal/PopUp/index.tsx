import React from 'react';
import {StyledPopUpCloseButton, StyledPopUpContent, StyledPopUpHeader, StyledPopUpWrapper} from './StyledComponents';

interface IProps {
  onClose: () => void;
}

const PopUp: React.FC<IProps> = ({onClose, children}): JSX.Element => (
  <StyledPopUpWrapper>
    <StyledPopUpHeader>
      You WIN!!!
    </StyledPopUpHeader>
    <StyledPopUpContent>
      {children}
    </StyledPopUpContent>
    <StyledPopUpCloseButton
      onClick={onClose}
      type='button'
    >
      Close
    </StyledPopUpCloseButton>
  </StyledPopUpWrapper>
);

export default PopUp;