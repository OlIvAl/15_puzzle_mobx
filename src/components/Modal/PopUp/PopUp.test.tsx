import * as React from 'react';
import PopUp from '.';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

interface IProps {
  onClose: () => void;
}

describe('PopUp', () => {
  const onClose = jest.fn();
  const CONTENT: string = 'test content';

  it('он отображается', () => {
    const wrapper = shallow<React.FC<IProps>>(
      <PopUp
        onClose={onClose}
      >
        {CONTENT}
      </PopUp>
    );

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('ссодержит, переданный в него, контент', () => {
    const wrapper = shallow<IProps>(
      <PopUp
        onClose={onClose}
      >
        {CONTENT}
      </PopUp>
    );

    expect(wrapper.contains(CONTENT)).toEqual(true);
  });

  it('при клике на кнопку вызовится onClose', () => {
    const wrapper = shallow<IProps>(
      <PopUp
        onClose={onClose}
      >
        {CONTENT}
      </PopUp>
    );

    wrapper.find('[type="button"]').simulate('click');

    expect(onClose).toHaveBeenCalled();
  });
});