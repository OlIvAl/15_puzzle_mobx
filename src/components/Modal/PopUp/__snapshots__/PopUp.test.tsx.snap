// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUp он отображается 1`] = `
<styled.div>
  <styled.div>
    You WIN!!!
  </styled.div>
  <styled.div>
    test content
  </styled.div>
  <styled.button
    onClick={[MockFunction]}
    type="button"
  >
    Close
  </styled.button>
</styled.div>
`;
